jf_agent/config_file_reader.py

```diff
@@ -96,7 +96,12 @@ def obtain_config(args) -> ValidatedConfig:
 
     run_mode = args.mode
     if run_mode not in VALID_RUN_MODES:
-        print(f'''ERROR: Mode should be one of "{', '.join(VALID_RUN_MODES)}"''')
+        agent_logging.log_and_print_error_or_warning(
+                logger,
+                logging.WARNING,
+                msg_args=[', '.join(VALID_RUN_MODES)],
+                error_code=2000,
+        )
```

Comment:
I think we're fine leaving this as a `print`, rather than converting it to a `log_and_print`, since it raises `BadConfigException` and that means there's basically no log output. This looks consistent with how we're handling other `BadConfigException` errors (like at line 118-119).